geospatial/src/Dem.cc

```diff
@@ -196,38 +240,59 @@ float Dem::MaxElevation() const
 }
 
 //////////////////////////////////////////////////
-void Dem::GeoReference(double _x, double _y,
+bool Dem::GeoReference(double _x, double _y,
     ignition::math::Angle &_latitude, ignition::math::Angle &_longitude) const
 {
+  if (this->dataPtr->isNonEarthDem)
+  {
+    ignerr << "Can not retrieve WGS84 coordinates from non-Earth DEM."
+            << std::endl;
+    return false;
+  }
+
   double geoTransf[6];
-  if (this->dataPtr->dataSet->GeoTransform(geoTransf) == CE_None)
+  if (this->dataPtr->dataSet->GetGeoTransform(geoTransf) == CE_None)
   {
     OGRSpatialReference sourceCs;
     OGRSpatialReference targetCs;
     OGRCoordinateTransformation *cT;
     double xGeoDeg, yGeoDeg;
 
     // Transform the terrain's coordinate system to WGS84
-    char *importString = strdup(this->dataPtr->dataSet->ProjectionRef());
+    const char *importString
+        = strdup(this->dataPtr->dataSet->GetProjectionRef());
```

Comment:
I'm testing this with a DEM that doesn't have a projection ref. I think we could catch this here instead of passing an empty string to `importFromWkt`. Something like this:

```cpp
    const char *importString
        = strdup(this->dataPtr->dataSet->GetProjectionRef());
    if (importString == nullptr || importString[0] == '\0')
    {
      ignwarn << "Projection coordinate system undefined." << std::endl;
      return false;
    }
```